feature/gribi/mpls_in_udp/otg_tests/mpls_in_udp_scale/mpls_in_udp_scale_test.go

```diff
@@ -0,0 +1,1470 @@
+		// Skip packets that are not UDP-over-IPv6. For debugging, we log only the first 5 skipped packets to avoid flooding the test logs if the capture contains many irrelevant packets (e.g., ARP, ICMP).
+		if udpLayer == nil || ipv6Layer == nil {
+			if packetCount <= 5 {
+				t.Logf("Packet %d: Skipping non-UDP-IPv6 packet", packetCount)
+			}
+			continue
+		}
+		mplsPacketCount++
+
+		// Validate IPv6 header
+		v6 := ipv6Layer.(*layers.IPv6)
+		if v6.DstIP.String() != pr.dstIP {
+			return fmt.Errorf("packet %d: got dstIP %s, want %s", packetCount, v6.DstIP, pr.dstIP)
+		}
+		if v6.SrcIP.String() != pr.srcIP {
+			return fmt.Errorf("packet %d: got srcIP %s, want %s", packetCount, v6.SrcIP, pr.srcIP)
+		}
+		if v6.HopLimit != pr.ipTTL {
+			return fmt.Errorf("packet %d: got hopLimit %d, want %d", packetCount, v6.HopLimit, pr.ipTTL)
+		}
+
+		// Extract UDP payload (MPLS header)
+		payload := udpLayer.LayerPayload()
+		if len(payload) < 4 {
+			return fmt.Errorf("packet %d: UDP payload too short (len=%d)", packetCount, len(payload))
+		}
+		mplsHeader := binary.BigEndian.Uint32(payload[:4])
+		label := (mplsHeader >> 12) & 0xFFFFF
+		bos := (mplsHeader >> 8) & 0x1
+		mplsTTL := mplsHeader & 0xFF
+
+		// Label validation
+		if len(labelMap) > 0 {
+			if _, ok := labelMap[uint64(label)]; !ok {
+				return fmt.Errorf("packet %d: got MPLS label %d, not in %v", packetCount, label, labelList)
+			}
+		} else if uint64(label) != pr.mplsLabel {
+			return fmt.Errorf("packet %d: got MPLS label %d, want %d", packetCount, label, pr.mplsLabel)
+		}
+		if bos != 1 {
+			return fmt.Errorf("packet %d: BOS bit = %d, want 1", packetCount, bos)
+		}
+		expectedMPLSTTL := innerTTL - 1
+		if uint32(mplsTTL) != expectedMPLSTTL {
+			return fmt.Errorf("packet %d: got MPLS TTL %d, want %d", packetCount, mplsTTL, expectedMPLSTTL)
+		}
+
+		validMplsPacketCount++
+		if validMplsPacketCount <= 2 {
+			t.Logf("Packet %d: MPLS validation PASSED", packetCount)
+		}
+	}
+
+	// Summary
+	t.Logf("=== PACKET CAPTURE VALIDATION SUMMARY ===")
+	t.Logf("Total packets: %d, UDP-IPv6: %d, Valid MPLS-in-UDP: %d",
+		packetCount, mplsPacketCount, validMplsPacketCount)
+
+	// Validation checks
+	switch {
+	case packetCount == 0:
+		return fmt.Errorf("no packets captured on port %s", otgPortName)
+	case mplsPacketCount == 0:
+		return fmt.Errorf("no UDP-IPv6 packets found on port %s", otgPortName)
+	case validMplsPacketCount == 0:
+		return fmt.Errorf("no valid MPLS-in-UDP packets found on port %s", otgPortName)
+	case validMplsPacketCount < mplsPacketCount/2:
+		return fmt.Errorf("too many invalid packets: %d/%d",
+			mplsPacketCount-validMplsPacketCount, mplsPacketCount)
+	}
+
+	t.Logf("Validation PASSED: %d valid MPLS-in-UDP packets", validMplsPacketCount)
+	return nil
+}
+
+// Main test entry point.
+func TestMPLSinUDPScale(t *testing.T) {
+	t.Helper()
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+	ctx := context.Background()
```

Comment:
Please use `t.Context()` inside tests.

https://google.github.io/styleguide/go/decisions#contexts

```suggestion
	ctx := t.Context()
```